### ISSUE TITLE
False positive of UnnecessaryApply when using `this`

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#1829]
```kotlin
PassengerSelectorDialogFragment().newInstance(it.copy()).apply {
  show(supportFragmentManager, this.tag)
}
```

get's flagged with 1.0.0

It should not though since I'm getting the `tag` from `this` which is only applicable when I'm using `apply`

**Comment by 3flex** [comment]
I just added this test to `UnnecessaryApplySpec` expecting it to fail but it passes:
```kotlin
            it("should not report when 'this' referenced more than once") {
                assertThat(subject.lint("""
                    PassengerSelectorDialogFragment().newInstance(it.copy()).apply {
                        show(supportFragmentManager, this.tag)
                    }
                """)).isEmpty()
            }
```

**Comment by vanniktech** [comment]
Weird. 1.0.0 fails for me on the code base. But then again, I'm no longer working on that code base so :man_shrugging:

**Comment by 3flex** [comment]
I'm going to close this then - if you can replicate or come across a similar example please reply here and we can take a look. Thanks!

**Comment by vanniktech** [comment]
Yo!
